data/base.go

```diff
@@ -0,0 +1,286 @@
+	timeZoneOffsetTag: fmt.Sprintf("TimezoneOffset needs to be in minutes and greater than %d and less than %d", tzValidationLowerLimit, tzValidationUpperLimit),
+}
+
+func PayloadValidator(v *valid.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+	//a place holder for more through validation
+	if _, ok := field.Interface().(interface{}); ok {
+		return true
+	}
+	return false
+}
+
+func AnnotationsValidator(v *valid.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+	//a place holder for more through validation
+	if _, ok := field.Interface().([]interface{}); ok {
+		return true
+	}
+	return false
+}
+
+func TimezoneOffsetValidator(v *valid.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+	if offset, ok := field.Interface().(int); ok {
+		if offset >= tzValidationLowerLimit && offset <= tzValidationUpperLimit {
+			return true
+		}
+	}
+	return false
+}
+
+func TimeObjectValidator(v *valid.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+	if timeObject, ok := field.Interface().(time.Time); ok {
+		return isTimeObjectValid(timeObject)
+	}
+	return false
+}
+
+func isTimeObjectValid(timeObject time.Time) bool {
+	return !timeObject.IsZero() && timeObject.Before(time.Now())
+}
+
+func TimeStringValidator(v *valid.Validate, topStruct reflect.Value, currentStructOrField reflect.Value, field reflect.Value, fieldType reflect.Type, fieldKind reflect.Kind, param string) bool {
+	if timeString, ok := field.Interface().(string); ok {
+		return isTimeStringValid(timeString)
+	}
+	return false
+}
+
+func isTimeStringValid(timeString string) bool {
+	var timeObject time.Time
+	timeObject, err := time.Parse(time.RFC3339, timeString)
+	if err != nil {
+		timeObject, err = time.Parse("2006-01-02T15:04:05", timeString)
+		if err != nil {
+			return false
+		}
+	}
+
+	return isTimeObjectValid(timeObject)
+}
+
+//Cast type for use in casting our incoming generic json data to the expected types that our data model uses
+type Cast struct {
+	err *Error
+}
+
+//NewCaster creates a Cast
+func NewCaster(err *Error) *Cast {
+	return &Cast{err: err}
+}
+
+//ToString will return the given data as a string or add an error to the cast obj
+func (cast *Cast) ToString(fieldName string, data interface{}) string {
+	if data == nil {
+		return ""
+	}
+	aString, ok := data.(string)
+	if !ok {
+		cast.err.AppendFieldError(fieldName, data)
+	}
+	return aString
+}
+
+//ToFloat64 will return the given data as a float64 or add an error to the cast obj
+func (cast *Cast) ToFloat64(fieldName string, data interface{}) float64 {
+	if data == nil {
```

Comment:
Same thing about nil not being the same thing as the type's default value.  This is relevant to all of the types below.